dev/PackageManager/API/PackageDeploymentResolver.cpp

```diff
@@ -260,13 +307,22 @@ winrt::hstring Microsoft::Windows::ApplicationModel::PackageDeploymentResolver::
     // Did we find what we're looking for?
     if (bestFitPackageFullName.empty())
     {
-        Common::Logging::DebugLog(std::format(L"PackageDeploymentResolver: No match ({})",
-                         criteria.get()));
+        (void)LOG_HR_MSG(MSIXPACKAGEMANAGER_E_PACKAGE_SCAN_FAILED,
```

Comment:
another return to LOG_HR_MSG - unless these are absolutely necessary, we should avoid them and just go with direct TraceLoggingWrite/OutputDebugString calls to avoid the FACILITY_ITF confusion that Pratik was originally trying to address.  I would think all calls to LOG_HR_MSG are not really error scenarios (they are logging after all), so could be avoided.

Reply:
ByDesign. The others were changed as they weren't Errors, they're Information that used WIL to log and thus had to be a FAILED(hr).

In this case it is an error. This reports the error details but the function returns empty string for not found. If it returned HRESULT this would be RETURN_HR(). We return empty string rather than throwing an exception as this isn't an exceptional case and exception flow up the stack would have be inappropriate.

Maybe in theory this could be rearchitected to throw but the ripple impact up the stack can get invasive so it's not worth the effort.